### ISSUE TITLE
"formatter produced unexpected output" with comma after assert

### ISSUE THREAD

**Issue by DanTup** [dart-lang/dart_style#892]
This was raised by @Akash98Sky by at https://github.com/Dart-Code/Dart-Code/issues/2267. I trimmed it down slightly to this:

```dart
class ConfigurationService {
  Configuration _configuration;

  void set configuration(Configuration configuration) :
    assert(configuration != null), _configuration = configuration;
}
```

It seems to trip up on the (invalid) `,` after the assert.

---

```shell
danny@Dannys-MacBook fmttest % dartfmt test.dart
```

Hit a bug in the formatter when formatting test.dart.
The formatter produced unexpected output. Input was:

```dart
class ConfigurationService {
  Configuration _configuration;

  void set configuration(Configuration configuration) :
    assert(configuration != null), _configuration = configuration;
}
```

Which formatted to:
```dart
class ConfigurationService {
  Configuration _configuration;

  void set configuration(Configuration configuration);
}
```

Please report at github.com/dart-lang/dart_style/issues.

**Comment by munificent** [comment]
This code has a syntax error. Setters can't have constructor initialization lists. :)

The failure here is that it looks like the parser doesn't report that error so dartfmt dutifully tries to format the AST and then gets confused when the result doesn't match the original (invalid) syntax.

Filed a parser issue here: https://github.com/dart-lang/sdk/issues/41121

**Comment by DanTup** [comment]
Aha, I see. Thanks for the info (+ raising parser issue)!
